syntax = "proto3";

package fixbug;

option cc_generic_services = true;

message ResultCode {
    int32 errcode = 1;
    bytes errmsg  = 2;
}


//登陆消息类型
message LoginRequest {
    bytes name = 1;  
    bytes pwd  = 2;  
}

//登陆响应消息类型
message LoginResponse {
    ResultCode result = 1;
    bool   success    = 3;
}

message GetFriendListsRequest {
    uint32 userid = 1;
}

message User {
    bytes name = 1;
    uint32 age = 2;
    enum Sex {
        MAN    = 0;
        WOMAN  = 1;
    }
    Sex sex    = 3;
}

message GetFriendListsResponse {
    ResultCode result         = 1;
    //列表类型
    repeated User friendList  = 2;
}

service UserServiceRpc {
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc GetFriendLists(GetFriendListsRequest) returns(GetFriendListsResponse);
}
